qiskit/providers/aer/openpulse/duffing_model_generators.py

```diff
@@ -0,0 +1,438 @@
+    drive_symbols = _str_list_generator(drive_symbol + '{0}', oscillators)
+    sorted_coupling_edges = coupling_graph.sorted_graph
+    # populate coupling strengths in sorted order (vertex indices are now also sorted within edges,
+    # so this needs to be accounted for when retrieving weights from coupling_dict)
+    coupling_strengths = [coupling_dict.get(edge) or coupling_dict.get((edge[1], edge[0])) for
+                          edge in sorted_coupling_edges]
+    coupling_symbols = _str_list_generator(coupling_symbol + '{0}{1}', *zip(*sorted_coupling_edges))
+    cr_idx_dict = coupling_graph.two_way_graph_dict
+
+    hamiltonian_dict = _duffing_hamiltonian_dict(oscillators=oscillators,
+                                                 oscillator_dims=oscillator_dims,
+                                                 oscillator_freqs=oscillator_freqs,
+                                                 freq_symbols=freq_symbols,
+                                                 anharm_freqs=anharm_freqs,
+                                                 anharm_symbols=anharm_symbols,
+                                                 drive_strengths=drive_strengths,
+                                                 drive_symbols=drive_symbols,
+                                                 ordered_coupling_edges=sorted_coupling_edges,
+                                                 coupling_strengths=coupling_strengths,
+                                                 coupling_symbols=coupling_symbols,
+                                                 cr_idx_dict=cr_idx_dict)
+
+    hamiltonian_model = HamiltonianModel.from_dict(hamiltonian_dict)
+
+    # construct the u_channel_lo list
+    u_channel_lo = _cr_lo_list(cr_idx_dict)
+
+    system_model = PulseSystemModel(hamiltonian=hamiltonian_model,
+                                    u_channel_lo=u_channel_lo,
+                                    qubit_list=oscillators,
+                                    dt=dt)
+
+    return system_model, cr_idx_dict
+
+
+# Helper functions for creating pieces necessary to construct oscillator system models
+
+
+def _duffing_hamiltonian_dict(oscillators,
+                              oscillator_dims,
+                              oscillator_freqs,
+                              freq_symbols,
+                              anharm_freqs,
+                              anharm_symbols,
+                              drive_strengths,
+                              drive_symbols,
+                              ordered_coupling_edges,
+                              coupling_strengths,
+                              coupling_symbols,
+                              cr_idx_dict):
+    """Creates a hamiltonian string dict for a duffing oscillator model
+
+    Note, this function makes the following assumptions:
+        - oscillators, oscillator_dims, oscillator_freqs, freq_symbols, anharm_freqs,
```

Comment:
As with above function assumption means `oscillators` is unnecessary

Reply:
Here oscillators is a list containing arbitrary indices.